tests/test_nginx.py

```diff
@@ -10,6 +11,15 @@ def test_nginx_welcome_page(auto_container, host):
     """test that the default welcome page is served by the container."""
     host_port = auto_container.forwarded_ports[0].host_port
 
-    assert "Welcome to nginx" in host.check_output(
-        f"curl -sf --retry 5 --retry-connrefused http://localhost:{host_port}/"
+    # Retry 5 times with quadratic backoff delay
+    @retry(
+        wait=wait_exponential(multiplier=1, min=4, max=10),
+        stop=stop_after_attempt(5),
     )
+    def check_reply():
+        resp = requests.get(f"http://localhost:{host_port}/", timeout=30)
```

Comment:
this can be 

```patch
+        assert "Welcome to nginx" in host.check_output(
+            f"curl -sf --retry 5 http://localhost:{host_port}/"
```

instead, which avoids the requests dependency (don#t really care either way)

Reply:
No preference either, @dcermak you pick.

Reply:
tests/test_ruby.py needs to be upated then with the same change.

Reply:
Let's first see what @dcermak picks.